cedar-policy-core/src/ast/value.rs

```diff
@@ -556,6 +542,48 @@ impl StaticallyTyped for ValueKind {
     }
 }
 
+/// Like `Display`, but optionally truncates embedded sets/records to `n`
+/// elements/pairs, including recursively.
+///
+/// `n`: the maximum number of set elements, or record key-value pairs, that
+/// will be shown before eliding the rest with `..`.
+/// `None` means no bound.
+///
+/// Intended for error messages, to avoid very large/long error messages.
+pub trait BoundedDisplay {
+    /// Write `self` to the writer `f`, truncating set elements or key-value
+    /// pairs if necessary based on `n`
+    fn fmt(&self, f: &mut impl std::fmt::Write, n: Option<usize>) -> std::fmt::Result;
```

Comment:
Nit: we can add two methods with default implementations `fmt_bounded` and `fmt_unbounded` so that users don't have to add `Some` or `None` as an argument.

Reply:
Sure done.